lib/puma/cluster.rb

```diff
@@ -181,7 +181,7 @@ def check_workers
         # we need to phase any workers out (which will restart
         # in the right phase).
         #
-        w = @workers.find { |x| x.phase != @phase }
+        w = @workers.find { |x| x.phase < @phase }
```

Comment:
This fixes a race condition bug that causes the tests I'm adding here to fail intermittently, both on master and this branch, but more frequently on the latter. `#check_workers` gets called in the `Cluster#run` loop, and when reforking, there could be a point in time where this method gets called between when worker 0's phase has been incremented, and the phase has been incremented. This results in worker 0 being returned by this find and causes it to receive a SIGTERM below just before a refork happens, which is not ideal. Seeing as that's a special case where the worker's phase could be greater than the target phase changing this ensures that we don't terminate worker 0 in that case while also ensuring workers are incremented to the right phase in an ordinary phased restart scenario.